pkg/cosignhelper/cosignverify.go

```diff
@@ -104,12 +108,19 @@ func (vo *CosignVerifyOptions) Verify(ctx context.Context, images []string) erro
 			return fmt.Errorf("parsing reference: %w", err)
 		}
 
+		remoteOptions := []remote.Option{remote.WithContext(ctx), remote.WithTransport(httpTrans)}
+
+		// Include WithAuthFromKeychain option for the registries requiring authentication
+		authenticatedRegistries := strings.Split(os.Getenv(constants.AuthenticatedRegistry), ",")
+		if utils.ContainsRegistry(authenticatedRegistries, ref.Context().Registry.RegistryStr()) {
+			remoteOptions = append(remoteOptions, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+		}
+
 		var arrErr []error
 		for _, verifier := range pubKeys {
 			co := &cosign.CheckOpts{
 				RegistryClientOpts: []ociremote.Option{
-					ociremote.WithRemoteOptions(remote.WithContext(ctx)),
-					ociremote.WithRemoteOptions(remote.WithTransport(httpTrans)),
+					ociremote.WithRemoteOptions(remoteOptions...),
```

Comment:
This actually fixes another issue as well because when we pass `WithRemoteOptions` twice like this, the second options overwrites the first one. So based on my debugging `remote.WithContext(ctx)` was never getting assigned.
```
ociremote.WithRemoteOptions(remote.WithContext(ctx)),
ociremote.WithRemoteOptions(remote.WithTransport(httpTrans))
```

So, I am using single `ociremote.WithRemoteOptions(remoteOptions...)` like this now.

Reply:
Nice! 👏